### ISSUE TITLE
Can't publish to queue created from publisher

### ISSUE THREAD

**Issue by SergeATX** [SpinGo/op-rabbit#91]
The following code will successfully create queue axchange and the right binding,
but will throw exception on publish.
May be it is worth mentioning, I am using a user with no access to default exchange.

```
**val publisherKey = Publisher.queue(topic(
  Queue.passive(
    Queue("queueName", durable = true, exclusive = false, autoDelete = false,
      arguments = Seq(
      ))),
  List("topicName"),
  Exchange.passive(Exchange.topic("exchangeName", true, false))))
(rabbitControl ?
  Message(write(msg), publisherKey)
).mapTo[ConfirmResponse]**

... setting up new channel AMQChannel(amqp://RABBITMQ_USER@127.0.0.1:5672/,4)
... confirmed-publisher-channel connected
... com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - access to exchange 'amq.default' in vhost '/' refused for user 'RABBITMQ_USER', class-id=60, method-id=40)
... received AmqpShutdownSignal(com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - access to exchange 'amq.default' in vhost '/' refused for user 'RABBITMQ_USER', class-id=60, method-id=40)): shutdown
... closing broken channel AMQChannel(amqp://wren_web_integration@127.0.0.1:5672/,4)
... asking for new channel
... disconnected
```

Thanks in advance.

**Comment by timcharper** [comment]
The output indicates that this is probably a permissions error.

```
... com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - access to exchange 'amq.default' in vhost '/' refused for user 'RABBITMQ_USER', class-id=60, method-id=40)
... received AmqpShutdownSignal(com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - access to exchange 'amq.default' in vhost '/' refused for user 'RABBITMQ_USER', class-id=60, method-id=40)): shutdown
```

**Comment by SergeATX** [comment]
Sure it is a problem with permissions. Message gets published to default exchange for whatever reason instead of the exchange that the user has permission to access. I saw in the sources that similar problem exists with publishing to retry queue and abandon queue.

**Comment by SergeATX** [comment]
Tim,
I figured out the problem a solved it for my use-case.

When you publish on a queue, even if it was defined through topic and has info about exchange and the binding, the op-rabbit will publish to default exchange with the queue name as routing key.

In my case I don't have access to default exchange, so I had to uses akka-rabitmq library to materialize the exchange with the queue binding and then publish to that exchange.
